src/opts.rs

```diff
@@ -89,6 +89,25 @@ pub fn enable_caching(enabled: bool) {
     debug_assert!(error >= 0);
 }
 
+/// Set the maximum data size for the given type of object to be considered
+/// eligible for caching in memory.  Setting to value to zero means that that
+/// type of object will not be cached.  Defaults to 0 for [`ObjectType::Blob`]
+/// (i.e. won't cache blobs) and 4k for [`ObjectType::Commit`],
+/// [`ObjectType::Tree`], and [`ObjectType::Tag`].
+pub fn set_cache_object_limit(kind: ObjectType, size: libc::size_t) {
```

Comment:
I think the function should be unsafe, since there are requirements of synchronizing thread-safe access to global mutable state.

Reply:
I think you're right. But from this view, isn't `enable_caching` also unsafe?

- https://github.com/libgit2/libgit2/blob/346930e2808eaf9d74d895b6efc3abd085086a33/src/libgit2/cache.c#L21
- https://github.com/libgit2/libgit2/blob/346930e2808eaf9d74d895b6efc3abd085086a33/src/libgit2/cache.c#L25